app/src/main/java/com/krachkovsky/it_anews/ui/fragments/NewsListFragment.kt

```diff
@@ -0,0 +1,220 @@
+
+            val layoutManager = LinearLayoutManager(
+                view.context, LinearLayoutManager.VERTICAL, false
+            )
+            recyclerList.adapter = adapter
+            recyclerList.layoutManager = layoutManager
+            recyclerList.addHorizontalSpaceDecoration(RECYCLER_ITEM_SPACE)
+            recyclerList.addOnScrollListener(scrollListener)
+
+            swipeRefreshList.setOnRefreshListener {
+                adapter.submitList(emptyList())
+                currentPage = 1
+                loadNews()
+                swipeRefreshList.isRefreshing = false
+            }
+
+        }
+    }
+
+    private val scrollListener = object : RecyclerView.OnScrollListener() {
+        override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+            super.onScrolled(recyclerView, dx, dy)
+
+            val layoutManager = recyclerView.layoutManager as LinearLayoutManager
+            val firstVisibleItemPosition = layoutManager.findFirstVisibleItemPosition()
+            val visibleItemCount = layoutManager.childCount
+            val totalItemCount = layoutManager.itemCount
+
+            val isNotLoadingAndNotLastPage = !isLoading && !isLastPage
+            val isAtLastItem = firstVisibleItemPosition + visibleItemCount >= totalItemCount
+            val isNotAtBeginning = firstVisibleItemPosition >= 0
+            val isTotalMoreThanVisible = totalItemCount >= QUERY_PAGE_SIZE
+            val shouldPaginate = isNotLoadingAndNotLastPage && isAtLastItem && isNotAtBeginning &&
+                    isTotalMoreThanVisible && isScrolling
+            if (shouldPaginate) {
+                loadNews()
+                isScrolling = false
+            }
+        }
+
+        override fun onScrollStateChanged(recyclerView: RecyclerView, newState: Int) {
+            super.onScrollStateChanged(recyclerView, newState)
+            if (newState == AbsListView.OnScrollListener.SCROLL_STATE_TOUCH_SCROLL) {
+                isScrolling = true
+            }
+        }
+    }
+
+
+    private fun loadNews() {
+        NewsService.newsApi.getAllNews(
+            page = currentPage,
+            pageSize = QUERY_PAGE_SIZE
+        )
+            .enqueue(object : Callback<NewsEverything> {
+                override fun onResponse(
+                    call: Call<NewsEverything>,
+                    response: Response<NewsEverything>
+                ) {
+                    if (response.isSuccessful) {
+                        val newList = adapter.currentList
+                            .dropLastWhile { it == PagingData.Loading }
+                            .plus(response.body()?.articles?.map { PagingData.Content(it) }
+                                .orEmpty())
+                            .plus(PagingData.Loading)
+                        adapter.submitList(newList)
+                        currentPage++
+                        Log.d("AAA onResponse", "Success ${response.body().toString()}")
```

Comment:
явно будет вызываться toString
```suggestion
                        Log.d("AAA onResponse", "Success ${response.body()}")
```